crates/router/src/core/payment_link.rs

```diff
@@ -27,15 +28,15 @@ pub async fn retrieve_payment_link(
     payment_link_id: String,
 ) -> RouterResponse<api_models::payments::RetrievePaymentLinkResponse> {
     let db = &*state.store;
-    let payment_link_object = db
+    let payment_link_config = db
         .find_payment_link_by_payment_link_id(&payment_link_id)
         .await
         .to_not_found_response(errors::ApiErrorResponse::PaymentLinkNotFound)?;
 
-    let status = check_payment_link_status(payment_link_object.fulfilment_time);
+    let status = check_payment_link_status(payment_link_config.fulfilment_time);
```

Comment:
what if this is none, use `unwrap_or` and then pass the default value. I see that `session_expiry` is being already calculated below, use the same value instead of checking this.